tests/integration/utils/network.py

```diff
@@ -29,41 +30,33 @@ def next_range(self, number_of_ports: int) -> "PortRangeInclusive":
 
         return PortRangeInclusive(next_start, next_end)
 
+    @contextmanager
+    def allocate_port(self) -> int:
```

Comment:
I decided to use a context manager because we need to release the ports back to the pool of available ones, otherwise we run out of ports to use

Reply:
Could the getting free port use binding to port 0, set it reusable and return the number?
Something like
```python
import socket
free_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
free_socket.bind('localhost', 0)
free_socket(socket.SOL_SOCKET, socket.SO_REUSE_ADDR, 1)
free_socket.getsockname() # returns tuple (addr, socket)
```

Reply:
I tried that in the past (different project), and it was a flaky solution. Behavior of REUSE_ADDR and REUSE_PORT is not compatible across OSs, and IIRC the application also needs to set the flag, otherwise we have issues with lingering sockets. but if you want we can give a try.

Reply:
I think we can assume that sockets are BSD enough. And also discard Windows behavior SO_REUSEADDR as an issue here, these are used only for tests. I think the idiom I propose is fairly standard practice to get free port.

Reply:
I was not referring to differences between BSD or Windows, but Linux and BSD. It seems that this is working ATM, I would prefer to skip that rabbit hole for now, that can be done on a follow up.